### PR TITLE
Piotrk39/test.add e2e tests for data changes sub page

### PR DESCRIPTION
**Pull request by Piotrk39**
[AB#39227](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/39227) 

## Describe your changes

Adds E2E test for data changes Sub-page:
- Updates all fields with a setup via API
- Validates all changes in the table by comparing old data of the registration with update object

## Checklist before requesting a code review

- [x] I have performed a self-review of my code
- [x] Adding tests is unnecessary/irrelevant
- [x] The changes do not touch the UI/UX
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from [our PR guidelines](https://github.com/global-121/121-platform/blob/main/docs/CONTRIBUTING.md#pull-request-guidelines)

## Portal preview-deployment

This PR does not have any preview deployments yet.

### CODE REVIEW

File: e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  await test.step('Verify data changes are displayed correctly', async () => {
+    // Loop through all the updated fields
+    for (const [fieldKey, fieldName] of Object.entries(dataFields)) {
+      await tableComponent.filterColumnByDropDownSelection({
+        columnName: 'Field changed',
+        selection: fieldName,
+      });
+      // Wait for table to load after filtering
+      await page.waitForTimeout(200); // Small wait to ensure table is ready since DOM is already loaded and we are not receiving any loading events
```
[the remark this answers is not in the document]
Comment (Piotrk39) [reply e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts]: Unfortunately there is nothing to wait there since the DOM is loaded and unfortunately only `waitForTimeout` was the valid option in this case. This is allowed only in edge case scenarios that I believe we witness here.

File: e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  const basePage = new BasePage(page);
+  const programMonitoring = new ProgramMonitoring(page);
+  const tableComponent = new TableComponent(page);
+
+  const programTitle = 'NLRC OCW program';
+
+  await test.step('Navigate to program`s monitoring page', async () => {
+    await basePage.selectProgram(programTitle);
+    await programMonitoring.navigateToProgramPage('Monitoring');
+    await programMonitoring.selectTab({ tabName: 'Data Changes' });
```
[the remark this answers is not in the document]
Comment (Piotrk39) [reply e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts]: Changed to double quotes for the whole sentence and correct apostrophe for the words

File: e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      });
+      // Validate that only one row is displayed after filtering
+      await tableComponent.validateWaitForTableRowCount({
+        expectedRowCount: 1,
+      });
+      // Get expected values with language conversion if needed
+      let expectedOldValue = registrationOCW4[fieldKey]?.toString() ?? '';
+      let expectedNewValue = dataUpdate[fieldKey]?.toString() ?? '';
+      // Convert language codes to display names for preferredLanguage field
+      if (fieldKey === 'preferredLanguage') {
```
Comment (jannisvisser) [inline e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts]: this can use the GenericRegistrationAttributes enum

File: e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+test("All elements of Monitoring's `Data Changes` sub-page display correct data for Registration", async ({
+  page,
+}) => {
+  const basePage = new BasePage(page);
+  const programMonitoring = new ProgramMonitoring(page);
+  const tableComponent = new TableComponent(page);
+
+  const programTitle = 'NLRC OCW program';
+
+  await test.step("Navigate to program's monitoring page", async () => {
```
[the remark this answers is not in the document]
Comment (Piotrk39) [reply e2e/portal/tests/ProgramMonitoring/DataChangesTabDisplaysCorrectData.spec.ts]: Changed
